hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotDiffManager.java

```diff
@@ -1397,31 +1393,39 @@ long generateDiffReport(
             DiffReportEntry entry =
                 SnapshotDiffReportOzone.getDiffReportEntry(DELETE, key);
             deleteDiffs.add(codecRegistry.asRawData(entry));
-          } else if (Arrays.equals(oldKeyName, newKeyName)) { // Key modified.
+          } else if (isDirectoryObject &&
+              Arrays.equals(oldKeyName, newKeyName)) {
             String key = resolveBucketRelativePath(isFSOBucket,
                 newParentIdPathMap, newKeyName);
             DiffReportEntry entry =
                 SnapshotDiffReportOzone.getDiffReportEntry(MODIFY, key);
             modifyDiffs.add(codecRegistry.asRawData(entry));
-          } else { // Key Renamed.
+          } else {
+            // Key Renamed.
+            String keyPrefix = getTablePrefix(tablePrefix,
+                (isDirectoryObject ? fsDirTable : fsTable).getName());
             String oldKey = resolveBucketRelativePath(isFSOBucket,
                 oldParentIdPathMap, oldKeyName);
-            String newKey = resolveBucketRelativePath(isFSOBucket,
-                newParentIdPathMap, newKeyName);
-            renameDiffs.add(codecRegistry.asRawData(
-                SnapshotDiffReportOzone.getDiffReportEntry(RENAME, oldKey,
-                    newKey)));
-
             // Check if block location is same or not. If it is not same,
             // key must have been overridden as well.
-            if (isObjectModified(snapshotDiffObject.getOldKeyName(),
-                snapshotDiffObject.getNewKeyName(),
-                snapshotDiffObject.isDirectory() ? fsDirTable : fsTable,
-                snapshotDiffObject.isDirectory() ? tsDirTable : tsTable)) {
+            boolean isObjectModified = false;
```

Comment:
No in the case of both renames and modified we might need to check. It cannot be exclusive thing.

Reply:
Consider these cases:
Object is Renamed, Object is also modified [ RENAME, MODIFIED]
Object is Renamed but with cyclic renames(No effective Rename), Object is not modified [RENAME]
Object is Renamed, Object is not modified [RENAME]
Object is not Renamed, Object is modified [MODIFIED]
Object is Renamed but with cyclic renames(No effective Rename), Object is modified [MODIFIED]

Reply:
We are ensuring that if both object is not modified and also not renamed. We update the modified entry time in OmKeyInfo if we have to use the snapdiff for that.

Reply:
I see it can't be simple if-else. But I still suggested to extract out` isObjectModified()` to a boolean variable.

```suggestion
            boolean isObjectModified = isObjectModified(
                keyPrefix + codecRegistry.asObject(oldKeyName, String.class),
                keyPrefix + codecRegistry.asObject(newKeyName, String.class),
                isDirectoryObject ? fsDirTable : fsTable,
                isDirectoryObject ? tsDirTable : tsTable);

            if (isObjectModified) {
              // Here, oldKey name is returned as modified. Modified key name is
              // based on base snapshot (from snapshot).
              modifyDiffs.add(codecRegistry.asRawData(
                  SnapshotDiffReportOzone.getDiffReportEntry(MODIFY, oldKey)));
            }

            if (!isObjectModified || !Arrays.equals(oldKeyName, newKeyName)) {
              String newKey = resolveBucketRelativePath(isFSOBucket,
                  newParentIdPathMap, newKeyName);
              renameDiffs.add(codecRegistry.asRawData(
                  SnapshotDiffReportOzone.getDiffReportEntry(RENAME, oldKey,
                      newKey)));
            }
          }
```

Reply:
done